File: app/db/models/customer.py
from __future__ import annotations

from datetime import datetime
from typing import Optional

from sqlalchemy import DateTime, Integer, String
from sqlalchemy.orm import Mapped, mapped_column
from sqlalchemy.sql import func

from app.db.base import Base


class Customer(Base):
    __tablename__ = "customers"

    id: Mapped[int] = mapped_column(Integer, primary_key=True, index=True)
    customer_name: Mapped[str] = mapped_column(String(255), nullable=False, index=True)
    contact_person: Mapped[Optional[str]] = mapped_column(String(100))
    contact_phone: Mapped[Optional[str]] = mapped_column(String(50))
    contact_address: Mapped[Optional[str]] = mapped_column(String(500))
    created_at: Mapped[Optional[datetime]] = mapped_column(
        DateTime(timezone=True), server_default=func.now()
    )
    updated_at: Mapped[Optional[datetime]] = mapped_column(
        DateTime(timezone=True), server_default=func.now(), onupdate=func.now()
    )

File: frontend/src/types/customer.ts
export interface Customer {
  id: number;
  name: string;
  contact_person?: string;
  contact_phone?: string;
  contact_address?: string;
  created_at?: string;
  updated_at?: string;
}

export interface CustomerCreateDTO {
  name: string;
  contact_person?: string;
  contact_phone?: string;
  contact_address?: string;
}

export interface CustomerUpdateDTO extends Partial<CustomerCreateDTO> {}
